File: Tema 03 - Web API/PetShelter/PetShelter.Domain/Fundraiser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PetShelter.Domain
{
    public class Fundraiser
    {
        public int Id { get; set; } 
        public string Name { get; set; }
        public Person Owner { get; set; }
        public decimal GoalValue { get; set; }
        public DateTime DueDate { get; set; }
        public FundraiserStatus Status { get; set; }
        public DateTime CreationDate { get; set; }
        public decimal DonationAmount { get; set; }
        public ICollection<Person>? Donors { get; set; }
        public Fundraiser(string name, Person owner, decimal goalValue, DateTime dueDate)
        {
            Name = name;
            Owner = owner;
            GoalValue = goalValue;
            DueDate = dueDate;
            if(DateTime.Compare(dueDate, DateTime.Now)>0)
            {
                Status = FundraiserStatus.Active;
            }
            else
            {
                Status = FundraiserStatus.Closed;
            }
            CreationDate = DateTime.Now;
            DonationAmount = 0;
            Donors=new List<Person>();
        }
    }
}

File: Tema 06 - React/petshelter.ui/src/Models/Fundraiser.ts
import type { Person } from "./Person";

export class Fundraiser {
    id:number;
    name: string;
    owner: Person;
    goalValue: number;
    dueDate: Date;
    status: string;
    creationDate: Date;
    donationAmount: number;
    donors: Person[]

    constructor(name: string,
        owner: Person,
        goalValue: number,
        dueDate: Date,
       // status: string,
       // creationDate: Date,
       // donationAmount: number,
        //donors: Person[]
        ) {
            this.id=Math.floor(Math.random() * 100000);
            this.name=name;
            this.owner=owner;
            this.goalValue=goalValue;
            this.dueDate=dueDate;
            this.creationDate=new Date()
            if(this.dueDate>this.creationDate)
            {
                this.status="Active";
            }
            else
            {
                this.status="Closed";
            }
            this.donationAmount=0;
            this.donors=[];
    }
    static createFundraiserWithAllAttributes(name: string,
        owner: Person,
        goalValue: number,
        dueDate: Date,
       status: string,
       creationDate: Date,
       donationAmount: number,
        donors: Person[]
        ):Fundraiser {
            let fundraiser=new Fundraiser(name, owner, goalValue, dueDate)
            fundraiser.creationDate=creationDate
            fundraiser.status=status
            fundraiser.donationAmount=donationAmount
            fundraiser.donors=donors
            return fundraiser;
    }
}
